src/stores/right-panel/RightPanelStoreIPanelState.ts

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { MatrixEvent } from "matrix-js-sdk/src/models/event";
+import { User } from "matrix-js-sdk/src/models/user";
+import { Room } from "matrix-js-sdk/src/models/room";
+import { RoomMember } from "matrix-js-sdk/src/models/room-member";
+import { VerificationRequest } from "matrix-js-sdk/src/crypto/verification/request/VerificationRequest";
+
+import { GroupMember } from "../../components/views/right_panel/UserInfo";
+import { RightPanelPhases } from "./RightPanelStorePhases";
+
+export interface IPanelState {
+    member?: RoomMember | User | GroupMember;
+    verificationRequest?: VerificationRequest;
+    verificationRequestPromise?: Promise<VerificationRequest>;
+    // group
+    groupId?: string;
+    groupRoomId?: string;
+    // XXX: The type for event should 'view_3pid_invite' action's payload
+    widgetId?: string;
+    space?: Room;
+    // treads
+    threadHeadEvent?: MatrixEvent;
+    initialEvent?: MatrixEvent;
+    isInitialEventHighlighted?: boolean;
+}
+export interface IPanelStateStored {
+    member?: RoomMember | User | GroupMember;
+    verificationRequest?: VerificationRequest;
+    verificationRequestPromise?: Promise<VerificationRequest>;
+    // group
+    groupId?: string;
+    groupRoomId?: string;
+    // XXX: The type for event should 'view_3pid_invite' action's payload
+    widgetId?: string;
+    space?: Room;
+    // treads
+    threadHeadEventId?: string;
+    initialEventId?: string;
+    isInitialEventHighlighted?: boolean;
+}
+
+export interface IPhaseAndState {
+    phase: RightPanelPhases;
+    state: IPanelState;
+}
+
+export interface IPhaseAndStateStored {
+    phase: RightPanelPhases;
+    state: IPanelStateStored;
+}
+
+export function convertToStoreRoom(cacheRoom: { isOpen: boolean, history: Array<IPhaseAndState> }):
+{ history: Array<IPhaseAndStateStored>, isOpen: boolean } {
```

Comment:
Please look around the code base, as I don't think we use a style like this... If the function definition is multi-line, then the opening brace should be on its own line. It would be better to define interfaces instead of these complex object types.